### PR TITLE
[component:agw] Handling AUTH Failure with Reason #71(ngKSI is alread…

### PR DESCRIPTION
**Pull request by ganeshg87**
…y in use)

Fix:
  1. In UE rejects AUTH failure Response with Reason #71, AMF reinitiate AUTH with
      restart AUTH procedure with new ngKSI
Test case :
1. Tested Registration  using UT/(ueRanSImulator)

Signed-off-by: ganeshg87 <ganesh.gedela@wavelabs.ai>

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/oai/tasks/amf/amf_authentication.cpp [hunk 1]
```diff
@@ -565,6 +565,96 @@ int amf_proc_authentication_complete(
...
+  /*	Stop Timer T3560 */
+#if 0
+		OAILOG_INFO(
+			LOG_NAS_EMM,
+			"Timer:  Stopping Authentication Timer T3560 with id = %d\n",
+			auth_proc->T3560.id);
+		stop_timer(&amf_app_task_zmq_ctx, auth_proc->T3560.id);
+		auth_proc->T3560.id = NAS5G_TIMER_INACTIVE_ID;
+		OAILOG_INFO(LOG_NAS_EMM, "Timer: After Stopping T3560 Timer\n");
+#endif
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/amf/amf_authentication.cpp]: remove unused code.

File: lte/gateway/c/oai/tasks/amf/amf_authentication.cpp [hunk 2]
```diff
@@ -565,6 +565,96 @@ int amf_proc_authentication_complete(
...
+  OAILOG_DEBUG(
+      LOG_NAS_AMF, "Authentication of the UE is Failed with Error : %d",
+      msg->m5gmm_cause.m5gmm_cause);
+  m5gmm_cause cause = (m5gmm_cause) msg->m5gmm_cause.m5gmm_cause;
+
+  switch (cause) {
+    case m5gmm_cause::NGKSI_ALREADY_IN_USE: {
+      // select a new ngKSI and send the same EAP-request message to the UE
+      amf_ctx->_security.eksi =
+          (amf_ctx->_security.eksi - 1) % (EKSI_MAX_VALUE - 1);
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/amf/amf_authentication.cpp]: This does not look correct. Should not this be `(amf_ctx->_security.eksi + 1) % (EKSI_MAX_VALUE + 1);`?

Comment (ganeshg87) [reply lte/gateway/c/oai/tasks/amf/amf_authentication.cpp]: yes.it was not correct.
my bad. it was not intentional. I have tested "(amf_ctx->_security.eksi + 1) % (EKSI_MAX_VALUE + 1);".

File: lte/gateway/c/oai/tasks/amf/amf_authentication.cpp (hunk 1 above)
Comment (ganeshg87) [reply lte/gateway/c/oai/tasks/amf/amf_authentication.cpp]: T3560 timer needs to be handled here. actually, the timer T3560  handling code was commented. (start T3560 code was commented)

Comment (ulaskozat) [reply lte/gateway/c/oai/tasks/amf/amf_authentication.cpp]: Then, please provide a context before the commented out code, e.g., when it is planned to be uncommented.

**Comment by ulaskozat** [comment]
@ganeshg87 @panyogesh is there a plan to update this PR?

File: lte/gateway/c/oai/tasks/amf/amf_authentication.cpp [hunk 3]
```diff
@@ -565,6 +565,96 @@ int amf_proc_authentication_complete(
...
+        "AMF-PROC - Failed to authenticate the UE due to NULL"
+        "ue_mm_context\n");
+    OAILOG_FUNC_RETURN(LOG_NAS_AMF, rc);
+  }
+
+  amf_ctx = &ue_mm_context->amf_context;
+  nas5g_amf_auth_proc_t* auth_proc =
+      get_nas5g_common_procedure_authentication(amf_ctx);
+
+  /*	Stop Timer T3560 */
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/amf/amf_authentication.cpp:618]: please add the comment about when the `#if 0` and `#endif` should/will be removed.

**Pull request merged**